src/graph.py

```
@@ -33,84 +36,151 @@ def filter_lists_by_threshold(filter_list, *other_lists, threshold=0.1):
+    """
+    json_files: List[Path] = []
+    base_path = Path(base_directory)
+
+    for json_file in base_path.rglob("*.json"):
+        # Skip files in subset directories
+        if "subset" in json_file.parts:
+            continue
+        json_files.append(json_file)
+
+    return json_files
+
+
+def create_output_directory(
+    json_file_path: Path, base_input_dir: Path, base_output_dir: Path
+) -> Path:
+    """
+    Create the output directory structure that mirrors the input structure.
+
+    Args:
+        json_file_path: Path to the JSON file
+        base_input_dir: Base input directory (e.g., 'src/results/faithfulness')
+        base_output_dir: Base output directory (e.g., 'src/results/graphs')
+
+    Returns:
+        Path object for the output directory
+    """
+    # Get the relative path from the base input directory
+    relative_path = json_file_path.relative_to(base_input_dir)
+
+    # Create the output path - preserve the entire directory structure
+    # including the input directory name (faithfulness/monitorability)
+    input_dir_name = base_input_dir.name
+    output_dir = base_output_dir / input_dir_name / relative_path.parent
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    return output_dir
 
-    with open(RAW_DATA_PATH, "r") as f:
-        data = json.load(f)
 
+def process_json_file(
+    json_file_path: Path, output_dir: Path, hint_taking_threshold: float = 0.1
+) -> None:
+    """
+    Process a single JSON file and generate all graphs.
+
+    Args:
+        json_file_path: Path to the JSON file to process
+        output_dir: Directory to save the graphs in
+        hint_taking_threshold: Threshold for hint taking (default 0.1)
+    """
+    print(f"Processing: {json_file_path}")
+
+    # Load data from JSON file
+    try:
+        with open(json_file_path, "r") as f:
+            data = json.load(f)
+    except Exception as e:
+        print(f"Error loading {json_file_path}: {e}")
+        return
```

All of these `try/except: return` blocks are extremely dangerous. I've seen this lead to undiscovered issues way too many times. We should not be swallowing errors like this, we should be fixing whatever is causing them to happen!